### ISSUE TITLE
Exit after 10 seconds. "ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane." 

### ISSUE THREAD

**Issue by Francewhoa** [ungoogled-software/ungoogled-chromium-flatpak#5]
### OS/Platform

Debian, Ubuntu, and derivatives

### Installed

Flatpak

### Version

Version 124.0.6367.118 (Official Build, ungoogled-chromium) (64 bits)

### Have you tested that this is not an upstream issue or an issue with your configuration?

- [ ] I have tried reproducing this issue in Chrome and it could not be reproduced there
- [ ] I have tried reproducing this issue in vanilla Chromium and it could not be reproduced there
- [ ] I have tried reproducing this issue in ungoogled-chromium with a new and empty profile using `--user-data-dir` command line argument and it could not be reproduced there
  - We were able to reproduce this challenge with 2 ungoogled-chromium on two different devices

### Description

Start with Terminal displays: ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.

<a name="reproduce"></a>

### How to Reproduce?

1. Install Ungoogle-Chromium (UC) version 124.0.6367.201-1 from https://flathub.org/apps/details/com.github.Eloston.UngoogledChromium

   * Notes: 
   
     * Release page for version 124.0.6367.201-1 at https://github.com/ungoogled-software/ungoogled-chromium/releases/tag/124.0.6367.201-1

1. UC works fine. No challenge.

1. Install [this Mesa OpenCL](https://packages.debian.org/bookworm/mesa-opencl-icd) `sudo aptitude install "mesa-opencl-icd"`

1. Wait for the next release of UC. Update UC to the new release 124.0.6367.207-1. This steps is the most important one. Because without an update, this challenge can not be reproduce.

   * Notes: 
   
     * Release page for version 124.0.6367.207-1 at https://github.com/ungoogled-software/ungoogled-chromium/releases

1. Using GNOME, start UC. It does open. It is important to start UC with GNOME. Otherwise this challenge can not be reproduce with starting UC from Terminal.

1. After roughly 10 seconds, UC exit. This is the challenge.

   * Notes: 

     * After UC exit above. There is an unusual process still open. `wrap`. Which never exit and uses memory. Each time you repeat the steps above. At this present step, an additional `wrap` process is added and remains open. And so on.

     * The same challenge can be reproduce many previous releases of UC. Maybe a few weeks back.

### Actual behaviour

Find steps to reproduce above

### Expected behaviour

Ungoogle-Chromium (UC) starts without error message and does not automatically exit after 10 seconds

### Relevant log output

Find steps to reproduce above

Per the steps above, if you start UC from Terminal, this challenge can not be reproduce. But if somehow the Terminal log is useful for your discovery, here is it below.

1. Using Terminal, start UC `/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=/app/bin/chromium --file-forwarding io.github.ungoogled_software.ungoogled_chromium @@u %U @@`

1. Terminal display those errors. This is the first challenge.

> Gtk-Message: 20:18:01.327: Failed to load module "canberra-gtk-module"
> Gtk-Message: 20:18:01.328: Failed to load module "canberra-gtk-module"
> [2:2:0515/201801.392909:ERROR:chrome_browser_cloud_management_controller.cc(161)] Cloud management controller initialization aborted as CBCM is not enabled. Please use the `--enable-chrome-browser-cloud-management` command line flag to enable it if you are not using the official Google Chrome build.
> [33:33:0515/201801.537874:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538043:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538190:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538272:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538368:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538448:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538542:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538624:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538701:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538743:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538807:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538847:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538909:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.538948:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539011:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539050:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539112:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539153:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539215:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539254:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539316:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539355:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539418:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539457:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539520:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539558:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539622:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539662:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539723:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539762:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539824:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539862:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539925:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.539968:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540030:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540069:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540131:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540170:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540232:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540272:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540336:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540375:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540437:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540475:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540537:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540575:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540636:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540674:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540737:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540776:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540837:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540876:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540939:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.540977:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541039:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541078:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541140:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541186:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541251:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541290:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541352:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541391:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541452:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541491:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541553:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541601:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541665:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541703:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541765:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541804:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541868:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541908:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.541970:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.542009:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.542072:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.542110:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.542173:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.542211:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.542273:ERROR:gbm_wrapper.cc(72)] Failed to get fd for plane.: Aucun fichier ou dossier de ce type (2)
> [33:33:0515/201801.542311:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: Aucun fichier ou dossier de ce type (2)
> [2:2:0515/201801.746994:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.ScreenSaver.GetActive: object_path= /org/freedesktop/ScreenSaver: org.freedesktop.DBus.Error.NotSupported: This method is not implemented
> [2:19:0515/201802.143827:ERROR:ev_root_ca_metadata.cc(162)] Failed to decode OID: 0
> [33:33:0515/201806.525379:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
> [33:33:0515/201806.527935:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
> [33:33:0515/201809.994185:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!

### Additional context

Debian 10 Bookworm
GNOME 43.9
Wayland

#### What we tried but did not work

* Temporarily deactivate graphic acceleration at `chrome://settings/system`

* Remove Ungoogled-Chromium (UC). But keep its configuration. Then reinstall UC.
   * `flatpak remove io.github.ungoogled_software.ungoogled_chromium`
   * `flatpak install flathub io.github.ungoogled_software.ungoogled_chromium`

#### IDU

@Francewhoa :) Note to myself IDU_7B9EF9S3

**Comment by rany2** [comment]
Do you still have this issue? I'm failing to reproduce it still :(

**Comment by rany2** [comment]
What I am doing is:

```
$ flatpak kill io.github.ungoogled_software.ungoogled_chromium
$ flatpak update -y --commit=bcb5eb85f21e995ae833f3be28e6b1271f1bdfe178bc6388844a5ded150494cc io.github.ungoogled_software.ungoogled_chromium
$ flatpak update -y io.github.ungoogled_software.ungoogled_chromium
$ flatpak run io.github.ungoogled_software.ungoogled_chromium
```

with `mesa-opencl-icd` installed but no dice...

**Comment by Francewhoa** [comment]
>Do you still have this issue? 

@rany2, yes

---------

>I'm failing to reproduce it still :(

Thanks for trying to reproduce :)

**Comment by Francewhoa** [comment]
@rany2 :) One question for you below

I'm not a developer. But as an end-user, my guess is that this is maybe related to the fact that all the devices who are affected by this challenge have multiple video card installed. Such as, but not limited to, multiple screens per device. Where some screens use different video card. So multiple video drivers needs to be installed at the same time. Including, but not limited to Mesa. 

In comparison, all the devices not affected by this challenge have only a single video card. Maybe just a coincidence?

In other words, maybe when starting ungoogled-chromium (UC), somehow UC automatically does not choose and use a video driver it supports?

**Question**. Using a Terminal command, is there a way to start UC but force it to use a specific video driver and ignore all the other video drivers? If yes, to narrow down the cause of this challenge, both me and the Ubertus.Org team could contribute trying forcing UC to start with each video driver in turn. Then share the results here.

**Comment by rany2** [comment]
@Francewhoa Unfortunately I don't have a good answer for that, I am not very familiar with how to fix this or if it's a UC bug at all. If it's at all possible, can you confirm that this issue occurs for you on `org.chromium.Chromium` in Flatpak? If it does then this issue should be redirected there and whatever fix that gets implemented on that Flatpak I'll add to the UC one.

If you want to test your theory on whether this is a GPU related problem, you could try to run with `--disable-gpu --disable-software-rasterizer`. The issue with that error you got is that Chromium has a lot of logspam and returns many errors that actually should be ignored, and I'm not sure if this is one of them or not.

**Comment by iam-TJ** [comment]
Found this report during a search for clues on a mostly unrelated but identical symptom issue. I thought I'd offer some insight into the warnings reported here, namely:
```
ERROR:gbm_wrapper.cc(74)] Failed to get fd for plane.: No such file or directory (2)
[3447575:3447575:0712/102134.254792:ERROR:gbm_wrapper.cc(257)] Failed to export buffer to dma_buf: No such file or directory (2)
```
It turns out that these are a by-product of the application checking for various Direct Rendering Interface (DRI) capabilities and specifically `DRM_IOCTL_PRIME_HANDLE_TO_FD` [0]

I used `strace` to capture everything `chromium` did, found where it writes these errors and then worked backwards to find likely causes of these errors. These `ioctl()` calls seem to be it on my host:
```
[pid 3444987] ioctl(22, DRM_IOCTL_RADEON_GEM_CREATE, 0x7ffcef3bc010) = 0
[pid 3444987] ioctl(22, DRM_IOCTL_RADEON_GEM_VA, 0x7ffcef3bbff0) = 0
[pid 3444987] ioctl(22, DRM_IOCTL_RADEON_GEM_SET_TILING, 0x7ffcef3bcc2c) = 0
[pid 3444987] ioctl(25, DRM_IOCTL_PRIME_HANDLE_TO_FD, 0x7ffcef3bcc4c) = -1 ENOENT (No such file or directory)
...
write(2, "[3444987:3444987:0712/101629.973"..., 121[3444987:3444987:0712/101629.973963:ERROR:gbm_wrapper.cc(74)] Failed to get fd for plane.: No such file or directory (2)
) = 121
```
With that last being the cause of the warnings. Kernel documentation [1] says this:

> GEM also supports buffer sharing with dma-buf file descriptors through PRIME. GEM-based drivers must use the provided helpers functions to implement the exporting and importing correctly. See ?. Since sharing file descriptors is inherently more secure than the easily guessable and global GEM names it is the preferred buffer sharing mechanism. Sharing buffers through GEM names is only supported for legacy userspace. Furthermore PRIME also allows cross-device buffer sharing since it is based on dma-bufs.

So I think @Francewhoa hypothesis that these are related to multi-GPU hosts may be correct where no actual PRIME exists. In my case there is a 4x DisplayPort AMD Radeon GPU driving 3 outputs.

[0] https://dri.freedesktop.org/docs/drm/gpu/drm-uapi.html#c.DRM_CAP_PRIME
[1]  https://www.kernel.org/doc/html/latest/gpu/drm-mm.html

**Comment by Francewhoa** [comment]
Hi @iam-TJ, @networkException, @PF4Public, @rany2  🙂

We found a permanent fix for this challenge. Simply deactivate WebGL. Then, restart ungoogled-chromium-flatpak. 

---------------

Below is the same permanent fix. But with details for those interested.

The cause of this challenge is that, by default, ungoogled-chromium-flatpak (UCF) activate WebGL. But some devices and or some browsers do not support, or do not use WebGL, or their WebGL is misconfigured by the end-user. In turn, when UCF starts, it tries to directly and or indirectly connect with WebGL. This fails. In turn, UCF crashes after a few seconds and is not usable. 

For those not familiar with WebGL, two of its main use are: 
   1. Spies and track your online activity using [WebGL fingerprints](https://browserleaks.com/webgl)
   1. Faster display of images, videos, and other items. There are many [alternatives to the default WebGL](https://alternativeto.net/software/webgl/?license=opensource). Many of those alternative have stronger privacy and stronger security.

After 11 months of searching and trying different options, we found that those steps permanently fix this challenge :

### Steps

1. Using ungoogled-chromium-flatpak (UCF)  go to this URL `chrome://flags`
1. Search for this keyword `Disable WebGL`
1. Set `Disable WebGL` to `Activated`
1. Fully close UCF. Including all its windows.
1. Start UCF. Test that WebGL is fully deactivated. To do so, go to https://browserleaks.com/webgl  Under `WebGL Support Detection` if the result is `False` then it is a success. You have now permanently resolved this challenge. UCF stops automatically exiting after 10 seconds.

-----------------

I updated my original steps to reproduce up above by adding this new first steps:
_Use a device that either do not support WebGL or is not configured to use WebGL or automatically blocks WebGL._

---------------

Alternatively, if you really need WebGL activated, this Extension "WebGL Fingerprint Defender" protect your privacy against WebGL spying on you at https://chromewebstore.google.com/detail/webgl-fingerprint-defende/olnbjpaejebpnokblkepbphhembdicik

**Comment by Francewhoa** [comment]
Hi @iam-TJ, @networkException, @PF4Public, @rany2 🙂

We found a second permanent and better fix for this challenge. Activate the Flatpak permission `Background` for Ungoogled-Chromium.

-----------------------

Below is the same permanent fix. But with details for those interested.

For those not familiar with Flatpak permission `Background`, this screenshot below shows how to use [Flatseal](https://flathub.org/en/apps/com.github.tchx84.Flatseal) to activate it for Ungoogled-Chromium.

<img width="1203" height="973" alt="Image" src="https://github.com/user-attachments/assets/7b960feb-67fd-4d63-9540-6ecab732af8d" />

Why is this resolution better than deactivated WebGL? Because with the Flatpak permission `Background` activated, you do not need to deactivate WebGL. Optionally, this Extension "WebGL Fingerprint Defender" protect your privacy against WebGL spying on you at https://chromewebstore.google.com/detail/webgl-fingerprint-defende/olnbjpaejebpnokblkepbphhembdicik

When you exit Ungoogled-Chromium, if the permission `Background`was activated, Ungoogled-Chromium is not fully exited. Part of it remain in memory. According to some people, this is a privacy and security risk. Because depending on your present Ungoogled-Chromium configuration and activated extensions, Ungoogled-Chromium could continue to spy on you and collect data. There is a configuration in Ungoogled-Chromium to always force a full exit. So that nothing permanently remain in memory.

The Ungoogled-Chromium versions roughly before 124.0.6367.118 did not require this permission  `Background`. But somehow, starting around after version 124.0.6367.118, it is required. This explains why sometime this challenge could be reproduced after updating Ungoogled-Chromium. But could not be reproduced with freshly installed Ungoogled-Chromium.
